test/services/keyVault/keyVault-secret-tests.js

```diff
@@ -0,0 +1,377 @@
+        updateSecretWithVersion,
+        deleteSecret,
+        getSecretReturnsNotFound,
+        function () {done();}
+      ]);     
+      
+    });
+  });
+
+  xdescribe('list', function() {
+    it('should work', function(done) {
+
+      var maxSecrets = LIST_TEST_SIZE;
+      var expected = {};
+      var zeroCount = Object.keys(expected).length;
+
+      function createManySecrets(next) {
+
+        var secretCount = 0;
+        var errorCount = 0;
+
+        function createASecret() {
+          client.setSecret(vaultUri, SECRET_NAME + (secretCount+1), { value: SECRET_VALUE }, function(err, secretBundle) {
+            if (err && err.code == 'Throttled') {
+              ++errorCount;
+              return setTimeout(createASecret, errorCount * 2500);
+            }
+            if (err) throw err;
+            errorCount = 0;
+            var secretId = KeyVault.parseSecretIdentifier(secretBundle.id).baseIdentifier;
+            expected[secretId] = secretBundle.attributes;
+            ++secretCount;
+            if (secretCount < maxSecrets) {
+              return createASecret();
+            }
+            next();
+          });
+        }
+
+        createASecret();
+      }
+
+      function listSecrets(next) {
+        var currentResult;
+        client.getSecrets(vaultUri, null, function(err, result) {
+          if (err) throw err;
+          //console.log('getSecrets: ' + JSON.stringify(result, null, ' '));
+          validateSecretList(result, expected);
+          currentResult = result;
+          if (currentResult.nextLink) {
+            return getNextSecrets();
+          }
+          next();
+          
+          function getNextSecrets() {
+            client.getSecretsNext(currentResult.nextLink, function(err, result) {
+              if (err) throw err;
+              validateSecretList(result, expected);
+              currentResult = result;
+              if (currentResult.nextLink) {
+                return getNextSecrets();
+              }
+              if (Object.keys(expected).length !== zeroCount) {
+                throw new Error('Not all secrets were returned: ' + JSON.stringify(Object.keys(expected), null, ' '));
+              }
+              next();
+            });
+          }
+          
+        });
+      }
+
+      function deleteSecrets(next) {
+
+        var secretNum = 1;
+
+        function deleteASecret() {
+          client.deleteSecret(vaultUri, SECRET_NAME+secretNum, function(err, secretBundle) {
+            if (err) {
+              console.info('Unable to delete secret: ' + JSON.stringify(err));
+            }
+            ++secretNum;
+            if (secretNum <= maxSecrets) {
+              return deleteASecret();
+            }
+            next();
+          });
+        }
+
+        deleteASecret();
+      }
+
+      series([
+        createManySecrets,
+        listSecrets,
+        deleteSecrets,
+        function() { 
+          if (!suiteUtil.isMocked) {
```

Comment:
I got undefined symbol. Where is it defined? Also nobody else seem to be using it. Perhaps my branch is outdated?